### PR TITLE
chore(Jekyll Jade plugin): support <?code-excerpt?>

### PR DESCRIPTION
**Pull request by chalin**
This adds support for <?code-excerpt?> processing instructions found in Jade markdown.

Staged at: https://angulardart-org-dev.firebaseapp.com/playground/jade-test

Factored out code-excerpt processing methods from the NgMarkdown plugin, placing it into a mixin. Both NgMarkdown and the Jade plugins use the mixin.

Followup to #638, and so contributes to
- #637 - in helping us transition from Jade to markdown
- #374 - [ng] fix makeExample stylePatterns

### CODE REVIEW

File: src/playground/jade-test.jade [hunk 1]
```diff
@@ -11,7 +11,7 @@ permalink: /playground/jade-test
   Title and code not tied to a file:
```
Comment (kwalrath) [inline src/playground/jade-test.jade]: Delete the "(Code excerpts...)" paragraph?

File: src/playground/jade-test.jade [hunk 2]
```diff
@@ -63,3 +63,14 @@ permalink: /playground/jade-test
   ```
   <my-app>Loading...</my-app>
   ```
+  ## Converted (manually)
```
Comment (kwalrath) [inline src/playground/jade-test.jade]: This section doesn't exactly match what's in the non-Jade version. Which one should change?

File: src/playground/jade-test.jade (hunk 1 above)
Comment (chalin) [inline src/playground/jade-test.jade]: done

File: src/playground/jade-test.jade (hunk 2 above)
Comment (chalin) [inline src/playground/jade-test.jade]: Thanks. I've made them consistent.

**Comment by chalin** [comment]
> I see that ** is treated as bold (in the transformers section of the pubspec). Can you fix that now?

Good catch!  Looking into it.

**Comment by chalin** [comment]
Fixed and redeployed to the staging server.

**Pull request merged**